pkgs/by-name/im/imsprog/package.nix

```
@@ -0,0 +1,77 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, libsForQt5
+, libusb1
+, pkg-config
+, makeWrapper
+, gnome
+, wget
+}:
+
+stdenv.mkDerivation rec {
+  pname = "imsprog";
+  version = "1.3.2";
+
+  src = fetchFromGitHub {
+    owner = "bigbigmdm";
+    repo = "IMSProg";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-serRHyxYp+HV1h/MtEn8LOD8DkmfOb3vR5Mcypaq6RM=";
+  };
+
+  postPatch = ''
+    substituteInPlace IMSProg_programmer/CMakeLists.txt \
+      --replace "set(UDEVDIR \"/usr/lib/udev\")" "" \
+      --replace "''${UDEVDIR}/rules.d" "$out/lib/udev/rules.d"
+
+    substituteInPlace IMSProg_programmer/other/99-CH341.rules \
+      --replace 'MODE="0660", GROUP="plugdev", TAG+="uaccess"' 'MODE="0666"'
+
+    substituteInPlace IMSProg_editor/ezp_chip_editor.cpp \
+      --replace "/usr/share/imsprog/IMSProg.Dat" "$out/share/imsprog/IMSProg.Dat"
+
+    substituteInPlace IMSProg_programmer/mainwindow.cpp \
+      --replace "/usr/share/imsprog/IMSProg.Dat" "$out/share/imsprog/IMSProg.Dat"
+
+    substituteInPlace IMSProg_editor/main.cpp \
+      --replace "/usr/share/imsprog/" "$out/share/imsprog/"
+
+    substituteInPlace IMSProg_programmer/main.cpp \
+      --replace "/usr/share/imsprog/" "$out/share/imsprog/"
+
+    substituteInPlace IMSProg_editor/other/IMSProg_editor.desktop \
+      --replace "Exec=/usr/bin/IMSProg_editor" "Exec=$out/bin/IMSProg_editor" \
+      --replace "Icon=/usr/share/pixmaps/chipEdit64.png" "Icon=$out/share/pixmaps/chipEdit64.png" \
+      --replace "Path=/usr/bin/" "Path=$out/bin/"
+
+    substituteInPlace IMSProg_programmer/other/IMSProg_database_update.desktop \
+      --replace "Exec=/usr/bin/IMSProg_database_update" "Exec=$out/bin/IMSProg_database_update" \
+      --replace "Icon=/usr/share/pixmaps/IMSProg_database_update.png" "Icon=$out/share/pixmaps/IMSProg_database_update.png" \
+      --replace "Path=/usr/bin/" "Path=$out/bin/"
+
+    substituteInPlace IMSProg_programmer/other/IMSProg.desktop \
+      --replace "Exec=/usr/bin/IMSProg" "Exec=$out/bin/IMSProg" \
+      --replace "Icon=/usr/share/pixmaps/IMSProg64.png" "Icon=$out/share/pixmaps/IMSProg64.png" \
+      --replace "Path=/usr/bin/" "Path=$out/bin/"
+  '';
+
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
+  buildInputs = [ libsForQt5.qt5.qtbase libusb1 ];
```

```suggestion
  nativeBuildInputs = [ cmake pkg-config makeWrapper ];

  buildInputs = [ libsForQt5.qtbase libusb1 ];

  strictDeps = true;
```

P.S.: why the `libsForQt5.qt5.qtbase` instead of simply `libsForQt5.qtbase`?